go/libraries/doltcore/sqle/enginetest/dolt_server_test.go

```diff
@@ -329,10 +329,78 @@ func startServer(t *testing.T) (*sqlserver.ServerController, sqlserver.ServerCon
 	return sc, serverConfig
 }
 
+func startServerWithDefaultConfigOnly(t *testing.T) (*sqlserver.ServerController, sqlserver.ServerConfig) {
+	dEnv := dtestutils.CreateTestEnv()
+	serverConfig := sqlserver.DefaultServerConfig()
+
+	sc := sqlserver.NewServerController()
+	go func() {
+		_, _ = sqlserver.Serve(context.Background(), "", serverConfig, sc, dEnv)
+	}()
+	err := sc.WaitForStart()
+	require.NoError(t, err)
+
+	return sc, serverConfig
+}
+
 func newConnection(t *testing.T, serverConfig sqlserver.ServerConfig) (*dbr.Connection, *dbr.Session) {
 	const dbName = "dolt"
 	conn, err := dbr.Open("mysql", sqlserver.ConnectionString(serverConfig)+dbName, nil)
 	require.NoError(t, err)
 	sess := conn.NewSession(nil)
 	return conn, sess
 }
+
+func TestDoltServerRunningUnixSocket(t *testing.T) {
```

Comment:
It would be good to assert that this code really is using the unix socket and not connecting over TCP. One easy way to do this is to ask your serverConfig struct for the ConnectionString, and then just assert that it has a unix socket in it.

Reply:
I think there are some other edge cases that we should be testing... 
* assert that the socket file is not used in specific configurations, such as when host or port are set.
* assert that the server fails to start up if there is already a file in the socket file path (and of course gives a helpful, actionable error message about how to proceed)

You might consider BATS for some of these higher-level, end-to-end tests.

Reply:
We should also make sure there's some indication to customers in the logs that the server is listening on the socket file. 

For example, this startup log message should mention the socket and socket file path:
```
Starting server with Config HP="localhost:1234"|T="28800000"|R="false"|L="info"
```